Analytics/resources/request_grouped.py

```diff
@@ -0,0 +1,180 @@
+import pandas as pd
+import json
+import numpy as np
+from models.sensor import Sensor
+from models.location import Location
+
+"""
+This function groupes sensor data having different temporal frequencies by grouping at 
+hourly intervals. The grouping is based on the median of all sensor values that fall within
+the hourly interval for three reasons: a) The purpose of this function is to provide the 
+frontend with a convenient form of visualising the data. as such, it is not used in any subsequent 
+analytics b) Median is more robust to outliers and c) Median preserves the original datatype format.
+
+@params:
+	data: a json object from request_for_data.get_attribute_data. 
+	per_sensor: A boolean flag of whether the reformormatted request returns data per individual sensor
+	or per attribute
+"""
+
+def request_grouped_data(data, per_sensor):
+	df = pd.read_json(json.dumps(data), orient = 'records')
+	harm_df = pd.DataFrame(columns=['Attribute_Name','Attribute_Table',
+                                        'Sensor_id', 'Timestamp', 'Value'])
+    ## Get the attributes
+	for attribute in range(len(df)):
+		attr = df.iloc[attribute].Attribute_Name
+		attr_table = df.iloc[attribute].Attribute_Table
+		attr_values = df.iloc[attribute].Attribute_Values
+		    
+
+		harm_df_temp = pd.DataFrame(columns=['Attribute_Name','Attribute_Table',
+		                                'Sensor_id', 'Timestamp', 'Value'])
+
+		harm_df_temp['Attribute_Name'] = [attr for i in range(len(attr_values))]
+		harm_df_temp['Attribute_Table'] = [attr_table for i in range(len(attr_values))]
+
+		harm_df_temp['Sensor_id'] = [i['Sensor_id'] for i in attr_values]
+		harm_df_temp['Timestamp'] = [i['Timestamp'] for i in attr_values]
+		harm_df_temp['Value'] = [i['Value'] for i in attr_values] 
+
+		### append to dataframe
+		harm_df = harm_df.append(harm_df_temp,ignore_index=True)
+
+		### set datetime index
+		harm_df['Timestamp'] = pd.to_datetime(harm_df['Timestamp'])
+		harm_df = harm_df.set_index('Timestamp')
+		harm_df.Value = harm_df.Value.astype(float)
+
+		if not per_sensor:
+			### Using the median to preserve the data type during groupby aggregation
+			harm_df = harm_df.groupby([pd.Grouper(freq='1H'), 'Attribute_Name']).median()
+			harm_df.reset_index(inplace=True)
+
+		else:
+			harm_df = harm_df.groupby([pd.Grouper(freq='1H'), 'Attribute_Name', 'Sensor_id']).median()
```

Comment:
Can we enable the API to specify `freq`? 

E.g. by adding:

- Weekly:`1W`
- Daily:`1D`
- Hourly:`1H` (as already implemented)
- Minutely:`1T`

Reply:
Yes that should be straighforward. I'll implement it